l3kernel/l3prop.dtx

```diff
@@ -832,80 +823,48 @@
 %   output one key--value pair for the property list; otherwise complain
 %   about a missing or extra~|=|.
 %    \begin{macrocode}
-\cs_new_protected:Npn \prop_set_from_keyval:Nn #1#2
+\cs_new_protected:Npn \prop_set_from_keyval:Nn #1
   {
-    \prop_clear:N \l_@@_internal_prop
-    \@@_from_keyval:n {#2}
-    \prop_set_eq:NN #1 \l_@@_internal_prop
-    \prop_clear:N \l_@@_internal_prop
+    \prop_clear:N #1
+    \prop_put_from_keyval:Nn #1
   }
 \cs_generate_variant:Nn \prop_set_from_keyval:Nn { c }
-\cs_new_protected:Npn \prop_gset_from_keyval:Nn #1#2
+\cs_new_protected:Npn \prop_gset_from_keyval:Nn #1
   {
-    \prop_clear:N \l_@@_internal_prop
-    \@@_from_keyval:n {#2}
-    \prop_gset_eq:NN #1 \l_@@_internal_prop
-    \prop_clear:N \l_@@_internal_prop
+    \prop_gclear:N #1
+    \prop_gput_from_keyval:Nn #1
   }
 \cs_generate_variant:Nn \prop_gset_from_keyval:Nn { c }
 \cs_new_protected:Npn \prop_const_from_keyval:Nn #1#2
   {
-    \prop_clear:N \l_@@_internal_prop
-    \@@_from_keyval:n {#2}
+    \prop_set_from_keyval:Nn \l_@@_internal_prop {#2}
     \tl_const:Nx #1 { \exp_not:o \l_@@_internal_prop }
     \prop_clear:N \l_@@_internal_prop
   }
 \cs_generate_variant:Nn \prop_const_from_keyval:Nn { c }
-\cs_new_protected:Npn \prop_put_from_keyval:Nn #1#2
+\cs_new_protected:Npn \prop_put_from_keyval:Nn
   {
-    \prop_set_eq:NN \l_@@_internal_prop #1
-    \@@_from_keyval:n {#2}
-    \prop_set_eq:NN #1 \l_@@_internal_prop
-    \prop_clear:N \l_@@_internal_prop
+    \bool_if:NTF \l__keyval_allow_blank_keys_bool
```

Comment:
I'm surprised that `l3doc` wouldn't complain here about a foreign internal being used.

Reply:
It probably does, but it's a warning.  Maybe that boolean should be `__kernel`?

Reply:
@PhelypeOleinik seems reasonable. Should I push here or open a new PR?